tmc-plugin/src/fi/helsinki/cs/tmc/ui/PreferencesUIFactory.java

```diff
@@ -76,24 +76,29 @@ public void showPreferencesDialog(final ActionListener dialogListener) {
         if (dialog != null) {
             throw new IllegalStateException("Preferences UI already visible");
         }
-        
+
         ActionListener closeListener = new ActionListener() {
             @Override
             public void actionPerformed(ActionEvent e) {
-                panel = null;
-                dialog = null;
+                destroyDialog();
                 dialogListener.actionPerformed(e);
             }
         };
-        
+
         DialogDescriptor descriptor = new DialogDescriptor(
                 panel,
                 "TMC Settings",
                 false,
                 NotifyDescriptor.OK_CANCEL_OPTION,
-                NotifyDescriptor.PLAIN_MESSAGE,
+                null,
```

Comment:
In the [API documentation](http://bits.netbeans.org/dev/javadoc/org-openide-dialogs/org/openide/DialogDescriptor.html#DialogDescriptor(java.lang.Object,%20java.lang.String,%20boolean,%20int,%20java.lang.Object,%20java.awt.event.ActionListener)) for the constructor, that option is specified as the "default button" for the dialog box, so the original parameter seemed incorrect. Changing it to null didn't seem to have any effect.
I had to add the `WindowListener` to get the close event, since closing the window with the X doesn't seem to trigger the `ActionListener` specified in the `DialogDescriptor` constructor.
I'n not sure how we can make sure that the objects set to null really get destroyed though.